Packages/io.chainsafe.web3-unity/Runtime/Scripts/UI/ConnectScreen.cs

```diff
@@ -58,28 +41,40 @@ private async Task TryConnect(ConnectionProvider provider)
         {
             try
             {
-                ShowLoading();
+                if (provider.DisplayLoadingOnConnection)
+                {
+                    ShowLoading("Establishing connection...");
+                }
 
                 await Web3Unity.Instance.Connect(provider);
             }
             catch (Exception e)
             {
-
-                if (!(e is TaskCanceledException))
+                if (e is not TaskCanceledException && e.InnerException is not TaskCanceledException)
                 {
                     DisplayError(
                         "Connection failed, please try again.");
+                    
+                    Debug.LogException(e);
```

Comment:
Because it's called `TryConnect`, which implies it will try to connect safely and won't throw. If we throw exception here then the game will crash when running outside the editor as there is no catch statement to handle the exception.
If you look at the call stack you'd see that `TryConnect` is being called in the click handler. User doesn't even have an option to handle this error.

Reply:
I don't think it'll crash it'll simply reset to the next frame I think it's okay to throw so devs can collect info if they ever run their own connection provider or ours was faulty

Reply:
Oh right we're logging it nvm 🤦‍♂️